### ISSUE TITLE
Cannot extract installer with setup data version 5.6.2

### ISSUE THREAD

**Issue by AJenbo** [alanwoolley/innoextract-android#1]
Hi, I work on [DevilutionX](https://github.com/diasurgical/devilutionX). We recently released a port for Android. Some users are unable to play the full version after buying Diablo from GOG since they do not have access to a computer.
So I was hoping that Innoextract would be a solution for them, but it appears that the version is a bit to old to handle the installer. We have tested the latest Windows and Linux version of Innoextract which seams to work fine.

What is the forcast for a future update to this nice app :)

P.s. it looks like it requires version 1.8 of Innoextract

The current 3.2.0 release unfortunately crashes when trying to extract the files.

**Comment by alanwoolley** [comment]
Great project! I have an update in progress, but the Android ecosystem is moving so quickly that I'm struggling to keep up! I don't want to commit to a timeline as this is dependent on my spare time, but hopefully it will be in the first half of 2022.

**Comment by AJenbo** [comment]
I can very much sympathies with that sentiment :) Let us know if we can help you some how.

**Comment by alanwoolley** [comment]
@AJenbo As mentioned before, I'm working on an update for InnoExtract 1.8+. I intend to support the majority of the newer features that Daniel has implemented upstream _eventually_, particularly those around games from GOG, but it would help to know what I should be prioritising in order to give you enough to work with.

As I don't own a copy of Diablo from GOG, can you provide me with the following:

1. How is it distributed by GOG. Is it a single .exe, or does it have multiple files?
2. The output of `innoextract -l <installer.exe>`
3. The output of `innoextract -i <installer.exe>`
4. The output of `innoextract `--list-languages <installer.exe>`

Thanks!

**Comment by AJenbo** [comment]
It's a single exe file. I'll try and get you the additional info.

**Comment by AJenbo** [comment]
Here we go
```
.\innoextract.exe -v
innoextract 1.9
Extracts installers created by Inno Setup 1.2.10 to 6.0.5
```

```
.\innoextract.exe -l '.\setup_diablo_1.09_hellfire_v2_(30038).exe'
Listing "Diablo + Hellfire" - setup data version 5.6.2 (unicode)
 - "Patch_rt.mpq" [en-US], "dx\Patch_rt.mpq" [en-US] (579 KiB)
 - "DIABDAT.MPQ" [en-US] (493 MiB)
 - "hellfire\hellfire.mpq" [en-US] (62.5 MiB)
 - "hellfire\hfvoice.mpq" [en-US] (36 MiB)
 - "hellfire\hfmusic.mpq" [en-US] (32.8 MiB)
 - "hellfire\hfmonk.mpq" [en-US] (35.9 MiB)
Done.
```
(cut down to just the relevant files since it's pretty long)

```
.\innoextract.exe -i '.\setup_diablo_1.09_hellfire_v2_(30038).exe'
Inspecting "Diablo + Hellfire" - setup data version 5.6.2 (unicode)

Languages:
 - en-US

GOG.com game ID is 1412601690

Setup is not passworded!

Done.
```

```
.\innoextract.exe --list-languages '.\setup_diablo_1.09_hellfire_v2_(30038).exe'
Inspecting "Diablo + Hellfire" - setup data version 5.6.2 (unicode)
 - en-US
Done.
```

**Comment by alanwoolley** [comment]
@AJenbo Just to let you know that I haven't forgotten about this. I've had to rethink a significant area in order to support newer versions of Android, but I'm on track to have a beta version with support for this Inno version shortly. I'll keep you posted.

**Comment by AJenbo** [comment]
Thanks that is really good to hear, we are about to make a new update to the app so that will probably bring more people requesting this :D

**Comment by alanwoolley** [comment]
I've published [v4.0.0-beta1](https://github.com/alanwoolley/innoextract-android/releases/tag/v4.0.0-beta1). I'm anticipating another few betas, but I'm keen to get some early feedback to see if you can now extract "setup_diablo_1.09_hellfire_v2_(30038).exe" and whether it's outputting the files you expect.

**Comment by AJenbo** [comment]
It worked :tada: 

![image](https://github.com/alanwoolley/innoextract-android/assets/204594/7a9e7f7c-f1b7-43ea-97cf-a7dc751e6ad8)
![image](https://github.com/alanwoolley/innoextract-android/assets/204594/8e146f8b-bbf1-4901-91e6-bf3c566c02c7)
![image](https://github.com/alanwoolley/innoextract-android/assets/204594/8744d760-0e69-4b77-93be-0b5240039626)

**Comment by AJenbo** [comment]
I had two users test it, they both report that it can successfully extract all the needed files, and also confirmed that it will report a crash at the end. I would say that is good enough to solve the issue users are having. So despite the few remaining issues I would suggest pushing it as a beta on Google Play as well.

Currently people have to download and side load a PC emulator, install Ubuntu, then type some command to do this if they do not have access to a computer :( So ANY solution that can be installed from Google Play is a step up in usability.

**Comment by alanwoolley** [comment]
Excellent, thanks for your help.

It's already on its way to Google Play - just pending approval etc.

**Comment by alanwoolley** [comment]
This is now generally available as v4.0.0, currently rolling out on the Play Store. I will be closing this issue. Many thanks for your help @AJenbo.

**Comment by AJenbo** [comment]
thank you this is huge :)
